### PR TITLE
ADCM - 2836 -  UI: Add modal window for upgrade operation

### PR DESCRIPTION
**Pull request by dbardin90**
https://tracker.yandex.ru/ADCM-2836

### CODE REVIEW

**Comment by kuhella** [comment]
@remizov-arena I am not sure, but might be it should be PR to ADCM-2835 branch? cause all necessary functionality of backend for this feature in that brunch

**Comment by dbardin90** [comment]
For upgrades with configs to work correctly, some fixes on the backend should be done

File: web/src/app/shared/components/upgrades/master/master.component.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if (this.model.upgrades.length === 1) this.choose(this.model.upgrades[0]);
+  }
+
+  choose(upgrade: IUpgrade) {
+    this.upgrade = upgrade;
+    this.show = this.service.spotShow(upgrade);
+  }
+
+  isDisabled(value: IValue) {
+    return value && ((value.hostmap && value.hostmap.noValid) || (value.config && !value.config.form?.valid));
```
Comment (remizov-arena) [inline web/src/app/shared/components/upgrades/master/master.component.ts:66]: may be `value?.hostmap?.noValid || !value?.config?.form?.valid`

File: web/src/app/shared/components/upgrades/master/master.component.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// limitations under the License.
+import { Component, ElementRef, EventEmitter, OnInit, ViewChild } from '@angular/core';
+import { DynamicComponent, DynamicEvent } from '@app/shared/directives/dynamic/dynamic.directive';
+import { BaseDirective } from '@app/shared/directives';
+import { IMasterData, IValue, MasterService, whatShow } from './master.service';
+import { IUpgrade } from "@app/shared/components";
+import { UpgradeParameters } from "@app/shared/components/upgrades/upgrade.directive";
+
+@Component({
+  selector: 'app-master',
```
Comment (remizov-arena) [inline web/src/app/shared/components/upgrades/master/master.component.ts:20]: What are differents with `web/src/app/shared/components/actions/master/master.component.ts` ?
And are identical selectors the norm?

Comment (dbardin90) [reply web/src/app/shared/components/upgrades/master/master.component.ts:20]: Component is used only in parent directories, so it won't be a problem.

File: web/src/app/shared/components/upgrades/master/master.component.ts (hunk 1 above)
Comment (dbardin90) [reply web/src/app/shared/components/upgrades/master/master.component.ts]: fixed

**Comment by dbardin90** [comment]
Upgrade for hc_acl was fixed, we can go full test

**Review by remizov-arena — APPROVED** [review]

**Pull request merged**